src/get-workspaces-paths.js

```diff
@@ -1,31 +1,69 @@
 'use strict';
 
-const execa = require('execa');
+const fs = require('fs-extra');
+const { promisify } = require('util');
+const glob = promisify(require('glob'));
+const jsYaml = require('js-yaml');
 const path = require('path');
 const readJson = require('./json').read;
 const readJsonSync = require('./json').readSync;
 
+function getWorkspaces(workspaceCwd, _2dFilesArray){
+  let _1dFilesArray = Array.prototype.concat.apply([], _2dFilesArray);
+
+  let uniqueFiles = [...new Set(_1dFilesArray)];
+
+  let packagePaths = uniqueFiles.map(file => path.join(workspaceCwd, file, 'package.json'));
+
+  for (let packagePath in packagePaths) {
+    if (!fs.existsSync(packagePath)) {
+      let index = packagePaths.indexOf(packagePath);
+
+      if (index > -1) {
+        packagePaths.splice(index, 1);
+      }
+    }
+  }
+
+  let workspacesPaths = packagePaths.map(workspace => path.relative(workspaceCwd, workspace).replace('/package.json', ''));
+
+  return workspacesPaths;
+}
+
+
 async function getWorkspacesPaths({
   workspaceCwd,
 }) {
   let workspacePackageJson = await readJson(path.join(workspaceCwd, 'package.json'));
 
   let { workspaces } = workspacePackageJson;
 
+  let packagesGlobs;
+
   if (!workspaces) {
-    workspaces = (await execa('pnpm', ['recursive', 'exec', '--', 'node', '-e', 'console.log(process.cwd())'], { cwd: workspaceCwd })).stdout
-      .split(/\r?\n/)
-      .map(workspace => path.relative(workspaceCwd, workspace));
+    packagesGlobs = jsYaml.load(
+      fs.readFileSync(`${workspaceCwd}/pnpm-workspace.yaml`, { encoding: 'utf-8' }),
```

Comment:
we should use async here, use path.join, and the encoding can be omitted because that's the default.

Reply:
good catch, updated